### DISCUSSION TITLE
Unable to build eclipse application repository or site with patch features

### DISCUSSION THREAD

**Discussion by GorakhSaste** [eclipse-tycho/tycho#3846] [Q&A]
This issue is based on https://bugs.eclipse.org/bugs/show_bug.cgi?id=389698

The feature patches are required to introduce new features, since to support the specific plugins to provide customers with the feature patches. unable to create the patch repository for patch feature using maven tycho.

I have created the sample application to reproduce this issue. (attached tycho-patch-feature-example.zip)
In this sample application, com.example.tycho.feature.patch contains feature.xml which will contain below code is added.

<import feature="com.example.tycho.feature" version="1.0.0.qualifier" patch="true"/>

[tycho-patch-feature-example.zip](https://github.com/eclipse-tycho/tycho/files/15322784/tycho-patch-feature-example.zip)

Patch feature is built with maven Tycho, but while create site with com.example.tycho.update.patch for the patch feature then below errors getting.

```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-p2-repository-plugin:4.0.6:assemble-repository (default-assemble-repository) on project com.example.tycho.update.patch.eclipse-repository: Cannot resolve dependencies of project com.example.tycho:com.example.tycho.update.patch.eclipse-repository:eclipse-repository:1.0.0-SNAPSHOT
[ERROR]  with context {osgi.os=win32, osgi.ws=win32, org.eclipse.update.install.features=true, osgi.arch=x86_64, org.eclipse.update.install.sources=true}
[ERROR]   Software being installed: com.example.tycho.update.patch.eclipse-repository raw:1.0.0.'SNAPSHOT'/format(n[.n=0;[.n=0;[-S]]]):1.0.0-SNAPSHOT
[ERROR]   Missing requirement: com.example.tycho.feature.patch.feature.group 1.0.0.202405090622 requires 'org.eclipse.equinox.p2.iu; com.example.tycho.feature.feature.group [1.0.0.qualifier,1.0.0.qualifier]' but it could not be found
[ERROR]   Cannot satisfy dependency: com.example.tycho.update.patch.eclipse-repository raw:1.0.0.'SNAPSHOT'/format(n[.n=0;[.n=0;[-S]]]):1.0.0-SNAPSHOT depends on: org.eclipse.equinox.p2.iu; com.example.tycho.feature.patch.feature.group [1.0.0,1.0.1): See log for details
[ERROR] -> [Help 1]
```
Our Project needs a patch feature mechanism support.
Please consider providing support for feature patching.

**Comment by laeubi** [comment]
See
- https://github.com/eclipse-tycho/tycho/issues/3839#issuecomment-2112647260
